mesonbuild/dependencies/pkgconfig.py

```diff
@@ -99,9 +99,10 @@ def _detect_pkgbin(cls, silent: bool, env: Environment,
         else:
             assert cls.class_pkgbin[for_machine] is None, 'for mypy'
             mlog.debug(f'Pkg-config binary for {for_machine} is not cached.')
+            has_libdir = bool(env.properties[for_machine].get_pkg_config_libdir())
             for potential_pkgbin in find_external_program(
                     env, for_machine, 'pkgconfig', 'Pkg-config',
-                    env.default_pkgconfig, allow_default_for_cross=False):
+                    env.default_pkgconfig, allow_default_for_cross=has_libdir):
```

Comment:
Can you remove this patch? It's not safe. A real pkg-config built for the cross machine will define the system include/library paths specific to that cross machine, and possibly the sysroot dir.

It's also possible to set via the PKG_CONFIG_SYSTEM_INCLUDE_PATH / PKG_CONFIG_SYSTEM_LIBRARY_PATH / PKG_CONFIG_SYSROOT_DIR environment variables, but it's not urgent because things *may* work fine without it -- or may not. If users specifically defines their own pkg-config for cross, we assume they know what they are doing, but I don't think we can automatically guarantee it.

Reply:
Note that this is not PKG_CONFIG_LIBDIR from env, but pkg_config_libdir from cross file properties. In that case it is pretty much certain that user wants to use pkg-config.

Reply:
Are you really saying that in a PR that makes env2mfile scrape it from env and put it in a machine file? :D

Reply:
Regardless of what env2mfile does, using system pkg-config when pkg_config_libdir is set in cross file is the right thing to do IMHO.

Now what's potentially unsafe is for env2mfile to add pkg_config_libdir property, but I think that's also the right thing to do.